### DISCUSSION TITLE
I will make a fork

### DISCUSSION THREAD

**Discussion by CicadaSeventeen** [SpiralLinux/SpiralLinux-project#166] [General]
I find you often can not catch up with issues here. Maybe you are kind of busy. I am going to make a fork of SpiralLinux and resolve of of the issue myself.
The change I make will be written in `readme`. I may not often submit PR here, but you can just merge it when you want.

**Comment by CicadaSeventeen** [comment]
@geckolinux

**Comment by geckolinux** [comment]
Hi @CicadaSeventeen ,

Of course anybody is welcome to fork the project. However, when dealing with open source projects, patience is a virtue. :-)

SpiralLinux is not my full-time job, and in fact it's not even a part-time job. One of the reasons that I created SpiralLinux based on Debian is due to the slow rate of change of Debian. I intend and expect SpiralLinux to be a low-maintenance project requiring infrequent releases due to the low amount of change in Debian. I respond to questions and bug reports as time permits, giving priority to major issues that were caused by some sort of major deficiency in the default configuration of SpiralLinux. Things related to user error, user questions, Debian bugs, Debian limitations, and/or unique hardware compatibility issues are not something that SpiralLinux can solve via the default configuration, so they are not a priority, although I'm happy to offer suggestions as time permits.

**Reply by CicadaSeventeen** [reply]
Thanks a lot. 

I have been dealing with some issues we talked about before. You can see [here](https://github.com/CicadaSeventeen/SpiralLinux-project) to see what I have done and have planed to do.  I will respect most of your ideas ( ie, I will not remove virtualbox packages ) in this repo so that you can merge it sometime if you want. 

I am much more free of time these year because I suffer from depression and leave school for some treatment., though sometimes I may be too depressed that I can not do any work. So when I am fine I often try to help with some projects or do my own ones. I mean, if you have some idea but do not have enough time to do so, you can just tell me and maybe I can do the job.

**Comment by geckolinux** [comment]
Thanks @CicadaSeventeen , I really do appreciate the good ideas as well as your availability. I'm sure that we'll be able to help each other despite our changing availability and circumstances. Best wishes for you and your health.

Regarding PRs, I think it would be easiest to just talk about the issues and planned changes and how to implement them. I don't really have the SpiralLinux repository set up for pull requests, as they are just simple tar.gz uploads. I don't consider myself a developer, I'm just an experienced Linux user and configurator, so I don't really need or want to implement the full git development workflow.

**Comment by geckolinux** [comment]
Regarding these suggestion changes from your Github:

> removing `/tmp` in btrfs subvol mount point
I don't think you really want to do that. Without a separate `/@tmp` subvolume the snapshots will not be bootable, and the contents of `/tmp` will be included in each `/` snapshot.

> adding dbus user session package
Which package name exactly would it be, and what would it accomplish?

Thanks a lot.

**Reply by CicadaSeventeen** [reply]
> Regarding these suggestion changes from your Github:
> 
> > removing `/tmp` in btrfs subvol mount point
> > I don't think you really want to do that. Without a separate `/@tmp` subvolume the snapshots will not be bootable, and the contents of `/tmp` will be included in each `/` snapshot.
> 
> > adding dbus user session package
> > Which package name exactly would it be, and what would it accomplish?
> 
> Thanks a lot.

1. I do not meet any problem without `/tmp` subvolume till now. I will test for some time remembering your suggestion. Why you think contents of `/tmp` should be preserved?
2. package name is `dbus-user-session`. It is necessary to use `snapd`, though I do not advice you to support `snapd` by default but to tell users not to use `snapd` if possble because it causes problem with `snapper`. Besides it is necessary to run `podman`. However, if I remember correctly, `apt install --install-recommends podman` or `apt install --install-recommends podman` will not recommend `dbus-user-sesion` (I will check it)

**Reply by geckolinux** [reply]
1. I just tested an installation of SpiralLinux without the `/@tmp` subvolume, and the snapshots from GRUB will not boot into the graphical interface.
     - > Why you think contents of /tmp should be preserved?
     
     The contents of `/tmp` should _not_ be preserved. Removing the `/@tmp` subvolume will cause Btrfs / Snapper to include a snapshotted version of  `/tmp` in every snapshot, because snapshots are taken of everything on the `/@` subvolume, and without a separate subvol `/tmp` would also be part of that.
2. OK, thanks,  we might be able to preinstall `dbus-user-session` if it doesn't cause any new problems.
     - > tell users not to use `snapd` if possble because it causes problem with snapper
     
     Could you please explain the problem?

**Reply by CicadaSeventeen** [reply]
> 1. I just tested an installation of SpiralLinux without the `/@tmp` subvolume, and the snapshots from GRUB will not boot into the graphical interface.
>    
>    * > Why you think contents of /tmp should be preserved?
>    
>    The contents of `/tmp` should _not_ be preserved. Removing the `/@tmp` subvolume will cause Btrfs / Snapper to include a snapshotted version of  `/tmp` in every snapshot, because snapshots are taken of everything on the `/@` subvolume, and without a separate subvol `/tmp` would also be part of that.
> 2. OK, thanks,  we might be able to preinstall `dbus-user-session` if it doesn't cause any new problems.
>    
>    * > tell users not to use `snapd` if possble because it causes problem with snapper
>    
>    Could you please explain the problem?

1. You are right. I mistakenly think Debian will automatically mount a  tmpfs on `/tmp` when `/tmp` is not mounted by `fstab`. This is behavior of Arch and Fedora, but not for Debian. I will undo this change. Thanks a lot.
2. `snap` mount read-only files on `/snap`, if it is not unmounted at the time you use  `snapper undochange` (I always prefer to use undochange rather than rollback, probably rollback with same problem), the read-only files will be stuck here. This can be solved by make `/snap` a subvolume. However, `snap` also put files into `/var/lib/snapd` (I do not know if any other places are used by snap). Although `/var/lib/snapd` will not cause problem for `snapped`, it will be inconsistent with `/snap` after rollback. You have to make it another sbuvolume to avoid that problem.
Do you remember openSUSE main repo does not include `snapd`? That is the point.

**Reply by geckolinux** [reply]
2) I see, thanks for the good explanation.
Have you tested a `sudo snapper-rollback X` ? (Note that this is _not_ the `snapper` command with a space between `snapper rollback` but rather this [snapper-rollback](https://github.com/jrabinow/snapper-rollback) utility.) From what you describe it seems like it should work to rollback with `snapper-rollback`, because `/snap` and `/var/lib/snap` would be together on the same `/@` subvolume snapshot, so they would be together in a consistent state.

**Comment by CicadaSeventeen** [comment]
Reports: Some of my changes that may help you.
1. Adding `fonts-noto`. It provides fonts for all unicode, otherwise some languages will show as blocks (called tofu usually). You can search in language options when install by calamares, and you can see them,
2. Adding `task-*-desktop` packages. This will help you to install recommends packages for a desktop environment. Packages that easily to forget , like `dbus-user-session` which I told you before, can be isntalled by this way.
3. Removing `hplip-gui`. This is just a hp office program, why you install it? Maybe because you find it provides a `/hp` under `/lib/firmware`? If I remember correctly, `/lib/firmware/hp` is just a empty directory.

**Reply by geckolinux** [reply]
1. Thanks, good suggestion. It's rather large at 12MB. And is `fonts-noto-cjk` needed too?
2. Hmm, I don't really like including `task-*` packages because they usually bring a lot of bloat. We can compare the amount of additional packages and size that the `task-*-desktop` packages bring to each edition to see if it's worth it or not.
3. HPLIP is unfortunately necessary for HP printer/scanner owners.

Thanks again for the suggestions.
